### ISSUE TITLE
Module cannot be used in nx monorepo

### ISSUE THREAD

**Issue by CrowhopTech** [mscdex/mmmagic#167]
Created a new `nx` monorepo and ran `npm install mmmagic` in the root of the monorepo. When trying to run a project that imports it, I get this error:

```
> nx run importer:build

✘ [ERROR] Could not resolve "../build/Release/magic"

    node_modules/mmmagic/lib/index.js:1:20:
      1 │ var Magic = require('../build/Release/magic');
```

The files `../build/Release/magic.node` and `../build/Release/magic.a` are present relative to the given file `index.js`, so I suspect it's something to do with how Nx is managing the dependency.

Any advice or fixes on how to get around this are much appreciated!

**Comment by mscdex** [comment]
You're probably better off asking whoever manages the "nx" project. I don't know anything about it.

**Comment by CrowhopTech** [comment]
Sounds good. Thanks! If I find anything I'll try to post it back here as well

**Comment by CrowhopTech** [comment]
I was finally able to nail this down with a two-part solution. First, the Nx-specific part was that we were searching in the `dist/apps` directory, which isn't the normal location relative to `node_modules`. I added a symbolic link of the `node_modules/mmmagic/build` directory in `dist/apps` to satisfy the relative import.

The other issue was the `require` statement: for some reason, the default one in the mmmagic `lib/index.js` was tripping up for me. I had to change it to this: `var Magic = require(require('path').join(__dirname, '..', 'build', 'Release', 'magic'));`. I forked the repo [here](https://github.com/CrowhopTech/mmmagic) with that tiny tweak for my own use. I'm not sure if there's anything I just configured wrong for that part or what, but it's finally working at least in local development!
